chart/tests/test_pgbouncer.py

```diff
@@ -168,6 +168,34 @@ def test_existing_secret(self):
             "secret": {"secretName": "pgbouncer-config-secret"},
         } == jmespath.search("spec.template.spec.volumes[0]", docs[0])
 
+    def test_stats_no_existing_secret(self):
+        docs = render_chart(
+            "TEST-PGBOUNCER-STATS",
+            values={
+                "pgbouncer": {"enabled": True},
+            },
+            show_only=["templates/pgbouncer/pgbouncer-deployment.yaml"],
+        )
+
+        assert {
+            "name": "pgbouncer-stats",
+            "secret": {"secretName": "TEST-PGBOUNCER-STATS-pgbouncer-stats"},
+        } == jmespath.search("spec.template.spec.containers[1].env", docs[0])
```

Comment:
Basically `env` is a list and you are comparing it to a `dict`. Try this:

```suggestion
        assert [{
            "name": "pgbouncer-stats",
            "secret": {"secretName": "TEST-PGBOUNCER-STATS-pgbouncer-stats"},
        }] == jmespath.search("spec.template.spec.containers[1].env", docs[0])
```

or, a little more resilient and how I'd go:

```suggestion
        assert {
            "name": "pgbouncer-stats",
            "secret": {"secretName": "TEST-PGBOUNCER-STATS-pgbouncer-stats"},
        } in jmespath.search("spec.template.spec.containers[1].env", docs[0])
```

Reply:
you have a lot of patience 🙏  thank you - i'll try tonight 😄